Patch-Source: https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/2138
---
From 0de9a98f7977988e4290e25992eb51fd34fd693c Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Tue, 1 Oct 2024 12:24:09 +0100
Subject: [PATCH] fwupd: Do not use API removed from fwupd 2.0.x

All the systemd-offline logic was unused, and so removed.
---
 plugins/fwupd/gs-fwupd-app.c    | 5 -----
 plugins/fwupd/gs-plugin-fwupd.c | 4 ----
 2 files changed, 9 deletions(-)

diff --git a/plugins/fwupd/gs-fwupd-app.c b/plugins/fwupd/gs-fwupd-app.c
index bad97420ab..e4376b2656 100644
--- a/plugins/fwupd/gs-fwupd-app.c
+++ b/plugins/fwupd/gs-fwupd-app.c
@@ -117,11 +117,6 @@ gs_fwupd_app_set_from_device (GsApp *app,
 	    )
 		gs_app_set_state (app, GS_APP_STATE_UPDATABLE_LIVE);
 
-	/* only can be applied in systemd-offline */
-	if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_ONLY_OFFLINE))
-		gs_app_set_metadata (app, "fwupd::OnlyOffline", "");
-
-
 	/* reboot required to apply update */
 	if (fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_REBOOT))
 		gs_app_add_quirk (app, GS_APP_QUIRK_NEEDS_REBOOT);
diff --git a/plugins/fwupd/gs-plugin-fwupd.c b/plugins/fwupd/gs-plugin-fwupd.c
index fd60490554..8e3d8ab2c8 100644
--- a/plugins/fwupd/gs-plugin-fwupd.c
+++ b/plugins/fwupd/gs-plugin-fwupd.c
@@ -1430,10 +1430,6 @@ gs_plugin_fwupd_install_async (GsPluginFwupd                      *self,
 	 * https://github.com/fwupd/fwupd/issues/5522. */
 	g_set_object (&self->app_current, app);
 
-	/* only offline supported */
-	if (gs_app_get_metadata_item (app, "fwupd::OnlyOffline") != NULL)
-		install_flags |= FWUPD_INSTALL_FLAG_OFFLINE;
-
 	gs_app_set_state (app, GS_APP_STATE_INSTALLING);
 
 	fwupd_client_install_async (self->client, data->device_id,
-- 
GitLab

